logger.go

```diff
@@ -0,0 +1,12 @@
+package rsync
+
+import "github.com/gokrazy/rsync/internal/log"
+
+// Logger logs messages.
+type Logger = log.Logger
+
+// SetLogger overrides the default logger to use in rsync.
+// This should be call from the very beggining of the program.
+func SetLogger(logger Logger) {
```

Comment:
Instead of a global SetLogger (here and in the internal/log package), the logger should be configured for the `*rsyncd.Server` instance. 

We could use the [functional options pattern](https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html), changing the signature to `rsyncd.NewServer(modules []Module, options ...Option)` and then offering a `WithLogger` option.

Reply:
yeah, I agree. But that change would be rather big. As I mentioned in the issue, there are many ad-hoc `log.Printf` call across the code base. A few examples:

- https://github.com/gokrazy/rsync/blob/de1cff573bd22217f6d5669b89a70f84c3a51829/internal/anonssh/anonssh.go#L61
- https://github.com/gokrazy/rsync/blob/1266b42a0090c0a958efa16c838876dbeff58101/internal/rsyncwire/wire.go#L80

Do we want to convert them in the same PR?

Reply:
As a first step, we could introduce the server option and make it set the internal/log logger. While there is still global state at play, at least users of the rsyncd package don’t modify global state directly. In subsequent steps, we can then pass the logger to more and more types and thereby contain/eliminate global state.

Reply:
Sounds good. What do you think of the logger interface in the pr? If we can agree on that, I can make the changes tomorrow

Reply:
The logger interface seems fine to me. Thanks for working on this :)

Reply:
Updated, PTAL